.github/workflows/main.yml

```diff
@@ -0,0 +1,44 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - master
+
+  push:
+    branches:
+      - master
+
+jobs:
+  lint:
+    name: Run lint tests
+    runs-on: ubuntu-latest
+    steps:
+      # Check out the repo.
+      - uses: actions/checkout@v2
+
+      # Initiate caching for node_modules.
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        id: cache-node-modules
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-${{ hashFiles('yarn.lock') }}-${{ hashFiles('package.json') }}
+
+      # Install the Yarn dependencies.
+      - uses: actions/setup-node@v1
+      - name: Install dependencies
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: yarn install
+
+      # Run the linting tests.
+      - run: yarn lint
+      - run: yarn prettier:check-all
+
+  deploy:
+    name: Placeholder job for web app deployment
+    runs-on: ubuntu-latest
+    needs: lint
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

Comment:
You might want to put deployment into a separate workflow, we typically separate our tests/deployment/linting for python packages, for example so they show up separately in PR/push checks.

Reply:
Oh wait, nevermind, I see they're showing up separately in the PR, so organization probably best left up to preferences of developers.

Reply:
I actually wanted to put this in a separate workflow, but I also want this not to run if the linting tests fail. Currently, there's no way to have a job depend on another job from a separate workflow.